src/proto/connection.rs

```diff
@@ -200,9 +202,12 @@ where
 
     /// Advances the internal state of the connection.
     pub fn poll(&mut self, cx: &mut Context) -> Poll<Result<(), proto::Error>> {
+        let span = tracing::trace_span!("connection", peer = %P::NAME);
```

Comment:
Should this be something like `connection::poll`? `connection` sounds *to me* like a span that measures the full connection...

Reply:
Good call. Perhaps we want to add a span to the connection struct, either instead of or in addition to this span.